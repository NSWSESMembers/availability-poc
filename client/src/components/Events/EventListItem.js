import React from 'react';
import PropTypes from 'prop-types';

import { ListItem } from '../List';

// this component is intended to be used to display basic information about an event in a
// <FlatList> and go to event detail view when tapped
const EventListItem = ({ event, urgent, onPress }) => (
  <ListItem
    title={event.name}
    subtitle={event.details}
    subtitleEllipsis
    icon={urgent ? 'exclamation-triangle' : 'bullhorn'}
    onPress={onPress}
    urgent={urgent}
  />
);

EventListItem.propTypes = {
  onPress: PropTypes.func.isRequired,
  event: PropTypes.shape({
    name: PropTypes.string.isRequired,
  }),
  urgent: PropTypes.bool,
};

export default EventListItem;
