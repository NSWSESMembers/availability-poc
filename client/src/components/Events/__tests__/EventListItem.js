import React from 'react';
import renderer from 'react-test-renderer';

import EventListItem from '../EventListItem';

const noop = () => {};

test('renders correctly', () => {
  const event = {
    id: 1,
    name: 'Name',
    details: 'Details',
  };
  const tree = renderer.create(
    <EventListItem
      event={event}
      onPress={noop}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
