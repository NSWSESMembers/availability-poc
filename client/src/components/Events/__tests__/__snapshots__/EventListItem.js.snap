// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 2,
          "elevation": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 0,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 6,
          "paddingBottom": 10,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "flex": 0,
            "fontWeight": "bold",
            "includeFontPadding": false,
            "marginRight": 8,
            "textAlignVertical": "top",
          }
        }
      >
        Name
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Object {
            "color": "#6B6B6B",
            "flex": 0,
            "fontSize": 12,
            "includeFontPadding": false,
            "textAlignVertical": "bottom",
          }
        }
      >
        Details
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingLeft": 8,
          "width": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 30,
            },
            Object {
              "alignSelf": "center",
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        ï‚¡
      </Text>
    </View>
  </View>
</View>
`;
